prowler/providers/kubernetes/services/core/core_service.py

```diff
@@ -93,6 +97,44 @@ def __list_config_maps__(self):
                 f"{error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
             )
 
+    def __list_nodes__(self):
+        try:
+            response = self.client.list_node()
+            self.nodes = []
+            for node in response.items:
+                node_model = Node(
+                    name=node.metadata.name,
+                    uid=node.metadata.uid,
+                    namespace=node.metadata.namespace
+                    if node.metadata.namespace
+                    else "cluster-wide",
+                    labels=node.metadata.labels,
+                    annotations=node.metadata.annotations,
+                    unschedulable=node.spec.unschedulable,
+                    node_info=node.status.node_info.to_dict()
+                    if node.status.node_info
+                    else None,
+                )
+                self.nodes.append(node_model)
+        except Exception as error:
+            logger.error(
+                f"{error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
+            )
+
+    def __in_worker_node__(self):
+        try:
+            hostname = socket.gethostname()
+            for node in self.nodes:
```

Comment:
Do you need to iterate for all the nodes?

Reply:
Yes, since the key is the UID and not the name

Reply:
yes but you can pick one if it is always the same. If it would not be the same we cannot do that this way.